File: 3238-find-the-number-of-winning-players/3238-find-the-number-of-winning-players.py
class Solution:
  def winningPlayerCount(self, n: int, pick: List[List[int]]) -> int:
    data = [Counter() for _ in range(n)]
    for pos, c in pick:
      data[pos][c] += 1    
    return sum(max(data[i].values() or [0]) > i for i in range(n))   

File: 3238-find-the-number-of-winning-players/3238-find-the-number-of-winning-players.js
/**
 * @param {number} n
 * @param {number[][]} pick
 * @return {number}
 */
var winningPlayerCount = function (n, pick) {
	const g = Array.from({ length: n }, () => new Array(12).fill(0));
	for (const [player, color] of pick) {
		g[player][color] = g[player][color] || 0;
		g[player][color]++;
	}
	let playerCount = 0;
	for (let i = 0; i < n; i++) {
		const sameColorBallCounts = Math.max(...g[i]);
		if (sameColorBallCounts > i) playerCount++;
	}
	return playerCount;
};
